src/Google.Api.Gax/Platform.cs

```diff
@@ -78,21 +105,23 @@ public sealed class Platform
 
         private static async Task<GcePlatformDetails> LoadGceDetails()
         {
-            // TODO: Respect METADATA_EMULATOR_HOST environment variable. Where is this documented?
+            // Check if emulator is in use by looking for an emulator host in environment variable
+            // METADATA_EMULATOR_HOST. This is the undocumented but de-facto mechanism for doing this.
+            var metadataEmulatorHost = Environment.GetEnvironmentVariable("METADATA_EMULATOR_HOST");
             const string metadataUrl = "http://metadata.google.internal/computeMetadata/v1?recursive=true";
             const string metadataFlavorKey = "Metadata-Flavor";
             const string metadataFlavorValue = "Google";
             const long maxContentLength = 512 * 1024; // Maximum allowed metadata size.
             try
             {
-                var httpRequest = new HttpRequestMessage(HttpMethod.Get, metadataUrl);
+                var httpRequest = new HttpRequestMessage(HttpMethod.Get, metadataEmulatorHost ?? metadataUrl);
```

Comment:
Thanks, this is now done.
It does only check it once, as LoadInstance() is only called once, from the Lazy s_instance static field.